chimney/src/main/scala/io/scalaland/chimney/internal/macros/TransformerMacros.scala

```diff
@@ -239,21 +241,19 @@ trait TransformerMacros extends MappingMacros with TargetConstructorMacros with
       To: Type
   ): Either[Seq[TransformerDerivationError], Tree] = {
 
-    From.valueClassMember
-      .map { member =>
-        Right {
-          mkTransformerBodyTree0(config.derivationTarget) {
-            q"$srcPrefixTree.${member.name}"
-          }
-        }
-      }
-      .getOrElse {
-        // $COVERAGE-OFF$
-        Left {
-          Seq(CantFindValueClassMember(From.typeSymbol.name.toString, To.typeSymbol.name.toString))
-        }
-        // $COVERAGE-ON$
-      }
+    val fromValueClassMember = From.valueClassMember.toRight(
+      Seq(CantFindValueClassMember(From.typeSymbol.name.toString, To.typeSymbol.name.toString))
+    )
+
+    for {
+      fromValueClassMember <- fromValueClassMember
+      fromValueClassMemberType = fromValueClassMember.returnType
+      fromMemberAccessTree = q"$srcPrefixTree.${fromValueClassMember.name}"
+      transformerBodyTree <- resolveRecursiveTransformerBody(fromMemberAccessTree, config)(
+        fromValueClassMemberType,
+        To
+      )
+    } yield mkTransformerBodyTree0(transformerBodyTree.target)(transformerBodyTree.tree)
```

Comment:
```suggestion
    } yield transformerBodyTree.tree
```
Returning just recursively derived tree should be sufficient in this case.

Reply:
@krzemin Thanks for the suggestions! Sorry for not delivering for too long- I managed to figure these ones myself, but didn't have much time to actually make a fix. But I'm just working on the fix right now, so hope today PR will be ready!